helm/hpcc/templates/_warnings.tpl

```diff
@@ -52,4 +52,79 @@ Pass in dict with root and warnings
+   {{- $hasResources := include "hpcc.checkResources" (dict "resources" $component.resources) -}}
+   {{- if eq $hasResources "false" -}}
+    {{- $_ := set $ctx "missingResources" (append $ctx.missingResources $component.name) -}}
+   {{- end -}}
+   {{- if not $component.cost -}}
+    {{- $_ := set $ctx "defaultCpuRateComponents" (append $ctx.defaultCpuRateComponents $component.name) -}}
+   {{- end -}}
+  {{- end -}}
+ {{- end -}}
+ {{- range $component := .root.Values.eclccserver -}}
+  {{- if not $component.disabled -}}
+   {{- $hasResources := include "hpcc.checkResources" (dict "resources" $component.resources) -}}
+   {{- if eq $hasResources "false" -}}
+    {{- $_ := set $ctx "missingResources" (append $ctx.missingResources $component.name) -}}
+   {{- end -}}
+   {{- if not $component.cost -}}
+    {{- $_ := set $ctx "defaultCpuRateComponents" (append $ctx.defaultCpuRateComponents $component.name) -}}
+   {{- end -}}
+  {{- end -}}
+ {{- end -}}
+ {{- range $component := .root.Values.thor -}}
+  {{- if not $component.disabled -}}
+   {{- $hasResources1 := include "hpcc.checkResources" (dict "resources" $component.managerResources) -}}
+   {{- $hasResources2 := include "hpcc.checkResources" (dict "resources" $component.workerResources) -}}
+   {{- $hasResources3 := include "hpcc.checkResources" (dict "resources" $component.eclAgentResources) -}}
+   {{- if or (eq $hasResources1 "false") (or (eq $hasResources1 "false") (eq $hasResources1 "false")) -}}
+    {{- $_ := set $ctx "missingResources" (append $ctx.missingResources $component.name) -}}
+   {{- end -}}
+   {{- if not $component.cost -}}
+    {{- $_ := set $ctx "defaultCpuRateComponents" (append $ctx.defaultCpuRateComponents $component.name) -}}
+   {{- end -}}
+  {{- end -}}
+ {{- end -}}
+ {{- if $ctx.missingResources -}}
+  {{ $warning := dict "source" "helm" "severity" "warning" -}}
+  {{- $_ := set $warning "msg" (printf "Cost calculation requires resources to be provided for %s: %s" ((len $ctx.missingResources)| plural "component" "components") ($ctx.missingResources|toStrings)) -}}
+  {{- $_ := set $ctx "warnings" (append $ctx.warnings $warning) -}}
+ {{- end -}}
+ {{- if $ctx.defaultCpuRateComponents -}}
+  {{ $warning := dict "source" "helm" "severity" "warning" -}}
+  {{- $_ := set $warning "msg" (printf "Default cpu cost rate is being used for %s: %s" ((len $ctx.defaultCpuRateComponents)| plural "Component" "Components") ($ctx.defaultCpuRateComponents|toStrings)) -}}
+  {{- $_ := set $ctx "warnings" (append $ctx.warnings $warning) -}}
+ {{- end -}}
+ {{- /* Warn if any storage planes uses defaults cost values */ -}}
+ {{- with (.root.Values.global.cost) -}}
+  {{- if and (eq .storageWrites 0.0500000000001) (and (eq .storageAtRest 0.0208000000001) (eq .storageReads 0.00400000000001)) -}}
+   {{- $_ := set $ctx "usingDefaultStorageCosts" "true" -}}
+  {{- end -}}
+ {{- end -}}
+ {{- $usingDefaultCost := $ctx.usingDefaultStorageCosts |default "false" -}}
```

Comment:
Is this necessary.  Could the following test be if $ctx.usingDefautlStorageCosts?

Reply:
No, it's not required. Not sure why I did that.